packages/web3-providers-ws/src/index.js

```diff
@@ -178,7 +178,7 @@ WebsocketProvider.prototype._onClose = function (event) {
 
     if (this.requestQueue.size > 0) {
         this.requestQueue.forEach(function (request, key) {
-            request.callback(errors.ConnectionClosedError(event));
+            request.callback(errors.ConnectionNotOpenError(event));
```

Comment:
On 1.x atm when someone tries to `send` over websockets and the provider doesn't really exist - a `connection not open on send` is fired. It's a common error that happens if you haven't started your client.

The provider-improvements branch adds request clearance on close and throws a different error for this case. 

Reverting to the original message here. WDYT?

Reply:
I've checked the old WS provider and it does throw in the ``error`` or ``close`` case the ``InvalidConnection`` error and with the error msg ``on WS`` which is actually not helpful. This means I would prefer to use the ``ConnectionClosedError`` and to provide our consumers more details about it.

Reply:
> it does throw in the error or close case the InvalidConnection

Have opened draft pr #3302 to show these tests pass - e.g produce ConnectionNotOpen on 1.x. 

How did you generate InvalidConnection? That would be a good case to add here.

Reply:
the ``InavlidConnection`` error does get triggered as soon as the private [_timeout](https://github.com/ethereum/web3.js/blob/1.x/packages/web3-providers-ws/src/index.js#L252) method does get called.